models/transmission.go

```diff
@@ -35,28 +36,40 @@ type Transmission struct {
 func (t Transmission) MarshalJSON() ([]byte, error) {
 	test := struct {
 		Timestamps
-		ID           *string              `json:"id"`
-		Notification Notification         `json:"notification,omitempty"`
-		Receiver     *string              `json:"receiver,omitempty"`
-		Channel      Channel              `json:"channel,omitempty"`
+		ID           string               `json:"id,omitempty"`
+		Notification *Notification        `json:"notification,omitempty"`
+		Receiver     string               `json:"receiver,omitempty"`
+		Channel      *Channel             `json:"channel,omitempty"`
 		Status       TransmissionStatus   `json:"status,omitempty"`
 		ResendCount  int                  `json:"resendcount"`
 		Records      []TransmissionRecord `json:"records,omitempty"`
 	}{
 		Timestamps:   t.Timestamps,
-		Notification: t.Notification,
-		Channel:      t.Channel,
+		ID:           t.ID,
+		Notification: &t.Notification,
+		Receiver:     t.Receiver,
+		Channel:      &t.Channel,
 		Status:       t.Status,
 		ResendCount:  t.ResendCount,
 		Records:      t.Records,
 	}
-	// Empty strings are null
-	if t.ID != "" {
-		test.ID = &t.ID
+
+	// this is a horrible, no good hack but is necessary because a ResendCount of 0 trips up the omitempty tag
```

Comment:
I want to call out this and ask if anyone has a better idea for how to accomplish this.